pywbem_mock/_dmtf_schema.py

```diff
@@ -0,0 +1,392 @@
+
+    @property
+    def schema_mof_filename(self):
+        """
+        :term:`string` defining the the path for the DMTF schema
+        mof top level file name which includes the pragmas that define
+        all of the files in the DMTF schema. This filename is of the
+        general form
+        """
+        return self._schema_mof_filename
+
+    def __str__(self):
+        """
+        Return a short string representation of the
+        :class:`~pywbem_mock.DMTFSchema` object for human consumption.
+        """
+
+        return '%s(schema_ver=%s, schema_dir=%s, url=%s)' % \
+               (self.__class__.__name__, self.schema_ver_str,
+                self.schema_dir, self._mof_zip_url)
+
+    def __repr__(self):
+        """
+        Return a string representation of the :class:`~pywbem_mock.DMTFSchema`
+        object that is suitable for debugging.
+
+        The properties and qualifiers will be ordered by their names in the
+        result.
+        """
+
+        return '{0}(schema_ver={1}, schema_dir={2})'.format(
+            self.__class__.__name__, self.schema_ver, self.schema_dir)
+
+    def _setup_dmtf_schema(self):
+        """
+        Install the DMTF schema from the DMTF web site if it is not
+        already installed.
+
+        Once the schema in installed in 'schema_dir', it is no longer touched if
+        this function is recalled.  Further, if the zip file is in the schema
+        directory, but there no mof subdirectory, has been created
+        the schema is unzipped without downloading the zip file from the
+        DMTF.
+
+        This allows the dmtf schema to be downloaded once and reused and further
+        allows a much smaller footprint for the schema for repositories, etc by
+        unzipping the schema zip file if it has not already been unzipped.
+
+        Raises:
+            IOError if the remote url cannot be opened.
+        """
+        first = [True]  # List allows setting variable from inner function
+
+        def if_first(msg):
+            """
+            Inner method prints msg if var 'first' is False and verbose is
+            True. This allows displaying steps that are actually executed.
+            """
+            if first:
+                if self.verbose:
+                    print("")
+            first[0] = False
+            if self.verbose:
+                print(msg)
+
+        if not os.path.isdir(self.schema_dir):
+            if_first("Creating directory for CIM Schema archive: %s" %
+                     self.schema_dir)
+            os.mkdir(self.schema_dir)
+
+        if not os.path.isfile(self._mof_zip_filename):
+            if_first("Downloading CIM Schema archive from: %s" %
+                     self._mof_zip_url)
+
+            try:
+                ufo = urlopen(self._mof_zip_url)
+            except IOError as ie:
+                os.rmdir(self.schema_dir)
+                raise ValueError('Schema url %s not found. Exception: %s' %
```

Comment:
The schema archive is not found at the url. The message says the url is not found.
Also, I suggest to simply let the exception string follow after a colon, without the "Exception" word, i.e. remove ". Exception".